InnerEye/ML/Histopathology/models/deepmil.py

```diff
@@ -84,6 +87,14 @@ def __init__(self,
         self.encoder = encoder
         self.num_encoding = self.encoder.num_encoding
 
+        if class_names is not None:
+            self.class_names = class_names
```

Comment:
you should probably check if the class names and n_classes are in sync.

Reply:
Thanks, I have now added a check